### DISCUSSION TITLE
Property linking and expanding properties

### DISCUSSION THREAD

**Discussion by da-sypha** [scalar/scalar#3108] [Ideas]
Absolutely love Scalar, looks amazing, great work! Have a few suggestions, couldn't seem to find these in the discussions or issues sections so thought to raise them (apologies if raised before):

- **Property linking**. You might want to direct a reader straight to a property instead of the start of the endpoint, especially if there are lots of properties. I think Redocly supports this and seen it in other APIs. For an example, see the [OpenAI API docs](https://platform.openai.com/docs/api-reference/audio/createSpeech), hover over a property name to see the link. 
- **Property expansion**. You might want to expand properties to a certain level by default, say if you have small objects within objects etc. [Redocly](https://redocly.com/docs/api-reference-docs/configuration/functionality) supports this in their config for schemas (schemaExpansionLevel) and examples (jsonSampleExpandLevel).

Thanks!

**Comment by hanspagel** [comment]
Both are awesome suggestions! I’ve created issues for it:

* https://github.com/scalar/scalar/issues/3110
* https://github.com/scalar/scalar/issues/3109

**Reply by da-sypha** [reply]
Fab, thanks Hans

**Comment by da-sypha** [comment]
@hanspagel, just checking back on the property linking request. Had a look at #3109 and it's closed in favour of #1847, which I think talks about linking models and this is more directly to properties.

Also, I provided the wrong example that doesn't actually directly link to properties, apologies. Here's the example linking to an actual property, which is really useful for us as we have endpoints with lots of properties, and the guides can point directly to the property:
https://developers.intercom.com/docs/references/rest-api/api.intercom.io/admins#path=email

Thanks!

,
